libs/model/src/policies/handleLaunchpadTrade.ts

```diff
@@ -43,18 +43,15 @@ export async function handleLaunchpadTrade(
     throw new Error('Unsupported chain');
   }
 
-  const trade = await models.LaunchpadTrade.findOne({
+  const web3 = new Web3(chainNode.private_url! || chainNode.url!);
+  const block = await web3.eth.getBlock(event.rawLog.blockHash);
+
+  await models.LaunchpadTrade.findOrCreate({
     where: {
       eth_chain_id: chainNode.eth_chain_id!,
       transaction_hash: event.rawLog.transactionHash,
     },
-  });
-
-  if (!trade) {
```

Comment:
It doesn't make sense to remove this conditional. We don't need to initialize Web3 + fetch the block if the trade exists.

Reply:
So the reason I had merged the find and create together is becuase there is a bug due to not having a transaction between the get and create sql statements. When this handler processes two of the same trades subsequently, it goes through the if statement the second time and throws an error because it has not finished creating it the first time.

We don't need to initialize web3 and fetch the block, but there are tradeoffs to the other solution. In the other solution, if we have:

```
start transaction
get trade
if (trade)
  web3.fetchBlcok
  create trade
end transaction
```

Then this will block the second invocation of the handler for the amount of time the web3 fetch block takes (which I am not sure how long it takes on prod). But the way I have it currently, we always fetch the block, but it is not held up on the transaction